lib/Extension/LanguageServerPhpCsFixer/Model/PhpCsFixerProcess.php

```diff
@@ -0,0 +1,114 @@
+<?php
+
+namespace Phpactor\Extension\LanguageServerPhpCsFixer\Model;
+
+use Amp\Process\Process;
+use Amp\Promise;
+use Phpactor\Extension\LanguageServerPhpCsFixer\Exception\PhpCsFixerError;
+use Psr\Log\LoggerInterface;
+use function Amp\ByteStream\buffer;
+use function Amp\call;
+use Throwable;
+
+class PhpCsFixerProcess
+{
+    public const EXIT_SOME_FILES_INVALID = 4;
+    public const EXIT_FILES_NEEDS_FIXING = 8;
+
+    /**
+     * @param array<string,string> $env
+     */
+    public function __construct(
+        private string $binPath,
+        private LoggerInterface $logger,
+        private array $env = [],
+    ) {
+    }
+
+    /**
+     * @param  string[] $options
+     *
+     * @return Promise<string>
+     */
+    public function fix(string $content, array $options = []): Promise
+    {
+        return call(function () use ($content, $options) {
+            /** @var Process */
+            $process = yield $this->run('fix', ...[...$options, '-']);
+
+            $stdin = $process->getStdin();
+            $stdin->write($content);
+            $stdin->end();
+
+            $stdout = yield buffer($process->getStdout());
+            $exitCode = yield $process->join();
+
+            if ($exitCode !== 0
+                && $exitCode !== self::EXIT_SOME_FILES_INVALID
+                && $exitCode !== self::EXIT_FILES_NEEDS_FIXING
+                && $exitCode !== (self::EXIT_SOME_FILES_INVALID | self::EXIT_FILES_NEEDS_FIXING)
+            ) {
+                throw new PhpCsFixerError(
+                    $exitCode,
+                    $process->getCommand(),
+                    yield buffer($process->getStderr()),
+                    $stdout
+                );
+            }
+
+            return $stdout;
+        });
+    }
+
+    /**
+     * @param string[] $options
+     *
+     * @return Promise<string>
+     */
+    public function describe(string $rule, array $options = []): Promise
+    {
+        return call(function () use ($rule, $options) {
+            /** @var Process */
+            $process = yield $this->run('describe', ...[...$options, $rule]);
```

Comment:
![image](https://user-images.githubusercontent.com/530801/215273373-02b11fe2-1282-46c2-a4e9-081b788a051d.png)

:point_up: this is a bit "meta", would it be possible to parse out the first sentence?

Reply:
it might also be good to have just `rule_name` (or identifier) as the diagnostic title, then (perhaps optionally?) the rule description.

Reply:
I think the "Description of braces rule" part was not intended in first place, and shouldn't appear any more.

I did amend the code to remove that part, as well as I did remove few extra content for things as fixing configuration (that shows some of configuration options of php-cs-fixer related to the rule) and fixing example (code snippets from before and after fixing)

Reply:
If you look on the screenshot provided in first comment the rule name in my editor displays after the rule description.

I did briefly checked other LSP servers and most of them don't provide rule name in content, but when they do, then they append it on the end. I left it for now without the name